### PR TITLE
chore: Payments distribution graph

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
payments distribution bar graph
image
image

## Motivation and Context
new graph added

## How did you test it?
graph and table view as expected, shown in screenshot attached in description

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/NewAnalytics/NewAnalyticsTypes.res [hunk 1]
```diff
@@ -13,7 +13,7 @@ type dimension = [
   | #authentication_type
 ]
 type status = [#charged | #failure]
-type metrics = [#payment_processed_amount | #payment_success_rate]
+type metrics = [#payment_processed_amount | #payment_success_rate | #payment_distribution]
```
Comment (sagarnaikjuspay) [inline src/screens/NewAnalytics/NewAnalyticsTypes.res:16]: its not a diff metric its uses `#payment_success_rate` metric itself, with a group by value

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsDistribution/PaymentsDistributionTypes.res [hunk 2]
```diff
@@ -0,0 +1,8 @@
+type queryData = PaymentsSuccessRate | Connector
+
+type categories = [#connector]
```
Comment (sagarnaikjuspay) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsDistribution/PaymentsDistributionTypes.res:3]: use can use `dimention` type from `NewAnalyticsTypes` both are the same

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessedTypes.res [hunk 3]
```diff
@@ -8,3 +8,5 @@ type paymentsProcessedCols =
   | Count
   | Amount
   | TimeBucket
+
+type categories = [#time_bucket]
```
Comment (sagarnaikjuspay) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessedTypes.res:12]: `TimeBucket` is already there as a variant u can use the same only, if it need to be converted to string maybe use a mapper function

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsSuccessRate/PaymentsSuccessRateTypes.res [hunk 4]
```diff
@@ -1,3 +1,5 @@
 type queryData =
   | PaymentSuccessRate
   | TimeBucket
+
+type categories = [#time_bucket]
```
Comment (sagarnaikjuspay) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsSuccessRate/PaymentsSuccessRateTypes.res:5]: here as well

Comment (gitanjli525) [reply src/screens/NewAnalytics/PaymentAnalytics/PaymentsSuccessRate/PaymentsSuccessRateTypes.res]: okay

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessedTypes.res (hunk 3 above)
Comment (gitanjli525) [reply src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessedTypes.res]: okay

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsDistribution/PaymentsDistributionTypes.res (hunk 2 above)
Comment (gitanjli525) [reply src/screens/NewAnalytics/PaymentAnalytics/PaymentsDistribution/PaymentsDistributionTypes.res]: okay

File: src/screens/NewAnalytics/NewAnalyticsTypes.res (hunk 1 above)
Comment (gitanjli525) [reply src/screens/NewAnalytics/NewAnalyticsTypes.res]: okay

**Review by sagarnaikjuspay — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**